tools/cli/src/opt/kanidm.rs

```diff
@@ -197,6 +197,25 @@ pub enum GroupAccountPolicyOpt {
         #[clap(flatten)]
         copt: CommonOpt,
     },
+    /// Sets the maximum number of entries that may be returned in a
+    /// search operation.
+    #[clap(name = "limit-search-max-results")]
+    LimitSearchMaxResults {
+        name: String,
+        maximum: u32,
+        #[clap(flatten)]
+        copt: CommonOpt,
+    },
+    /// Sets the maximum number of entries that are examined during
+    /// a partially indexed search. This does not affect fully
```

Comment:
We need to document the difference between "fully" and "partially" indexed somewhere, but that can get done in another task.

Reply:
I actually hesitated about adding this one, but it's needed in conjunction with search max results. I also thought about having it automatically set to a value, or hiding it, but it all gets weird. 

Generally we've hidden a lot about the way the search internals work, because realistically, the defaults have been "so good" no one has thought to ask about them! 

Anyway, fully and partial indexed is highlighted as an error in the logs if it occurs and you hit the filter test max limit.